apollo-federation/src/operation/mod.rs

```diff
@@ -259,6 +261,12 @@ impl PartialEq for SelectionSet {
 
 impl Eq for SelectionSet {}
 
+impl Hash for SelectionSet {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.selections.hash(state);
```

Comment:
should this hash include the parent type of the selection set?

Reply:
Oh, we don't check the type position in `Eq` :cowboy_mild_panic:

Reply:
Yeah, this was the behavior in JS QP and AFAIK we rely on this behavior in other parts of the QP. The general idea here is that when we hash the selections we use that top level parent type when hashing and rely on the fact that all subselections then have to match the types (as at this point this op is within single subgraph)